pyonfx/shape.py

```diff
@@ -252,13 +252,13 @@ def bounding(self) -> Tuple[float, float, float, float]:
         """
 
         # Bounding data
-        x0: float = None
-        y0: float = None
-        x1: float = None
-        y1: float = None
+        x0: Optional[float] = None
```

Comment:
We should not provide linting for variables users can't see, right?

Reply:
This is true that since users can't see them, it's not really useful but having linting even in internal functions helps a lot for contributors like me because we don't necessarily know each type of each variable

Reply:
You have a good point. But, on the other side, linting can also complicate code, leaving too much informations in sight.
I propose to review variables' name and choose more appropriate names (whether any of them is not explanatory enough), while not adding linting: most of the time you can get a hunch of types just by reading it.

Reply:
I don't understand how linting can complicate the code. It helps developpers to have an idea of what a variable is and/or should accept. 
The Zen of Python said it: "Explicit is better than implicit". Linting isn't complex.

It also shows the errors before you have to run the code which is extremely powerful.
Following your idea, whe should do this:
```py
# self.styles = {}
self.dict_of_styles = {}
```
while just adding typehint resolve all our problems.

Reply:
I am of the idea that, whichever decision we take, we should keep it and propagate it to the entire code, so multiple cases should be considered.

I can provide you two examples where typehint can become really annoying:
```python3
# CASE 1
x: int = 1 # I can already clearly see x is an integer from the assignment
# CASE 2
# Immagine adding typehints to this already difficult to read function: https://github.com/CoffeeStraw/PyonFX/blob/master/pyonfx/shape.py#L421-L447
```
Sometimes the type of the variable is really obvious, sometimes really make reading the code harder. Not to mention that some types can really become complicated, e.g. the type of `x = [[[[[[[1,2]]]]]]]`.

Specifically for this project, even if generally I'm for typehinting, there are cases where I'm reclutant.

Reply:
I mean, the case 1 isn't annoying until you want to see your static linting checker to yell at you because you want to perform mathematical operations:
```py
x: int = 1
x /= 2  # linters will yell at you because x become float
```
About the case 2, I could generalise it as if it's hard to read the code without typehint, the code should probably refactorised... Also, Black formatting doesn't help in my opinion.